### PR TITLE
WIP supporting mvt

### PR DESCRIPTION
**Pull request by wheresrhys**

### CODE REVIEW

File: server/lib/allocate-tests.js [hunk 1]
```diff
@@ -27,11 +27,21 @@ module.exports = function(tests, user) {
 	});
 
 	const allocation = testsInRange.map(function (test) {
-		let rng = seedrandom(user.uuid + test.name);
-		let group = (rng() > 0.5) ? 'off' : 'on';
+		const variants = (test.abTestSetup && test.abTestSetup.variants) ? test.abTestSetup.variants.concat(['default']) : ['variant', 'default'];
```
Comment (commuterjoy) [inline server/lib/allocate-tests.js]: 'default' should be called 'control' imho

Comment (wheresrhys) [inline server/lib/allocate-tests.js]: I coulda sworn you preferred default when we spoke the other day. I'm all for control

File: server/lib/allocate-tests.js [hunk 2]
```diff
@@ -27,11 +27,21 @@ module.exports = function(tests, user) {
...
-		let group = (rng() > 0.5) ? 'off' : 'on';
+		const variants = (test.abTestSetup && test.abTestSetup.variants) ? test.abTestSetup.variants.concat(['control']) : ['variant', 'control'];
+		let index = Math.floor(seedrandom(user.uuid + test.name)() * variants.length);
+		let group = variants[index];
 
 		// Metrics: Track A/B allocation
-		metrics.count('allocation.' + test.name + '.' + group, 1);
+		metrics.count(`allocation.${test.name}.${group}`, 1);
+
+		// HACK: Track legacy allocations for a little while to avoid healthchecks going insane
```
Comment (tom-parker) [inline server/lib/allocate-tests.js]: Do we need to do this? Can we just let them go off and then fix them?

**Comment by tom-parker** [comment]
:+1:  Looks good otherwise

File: server/lib/allocate-tests.js (hunk 2 above)
Comment (wheresrhys) [inline server/lib/allocate-tests.js]: as I'm messing with flags I want to keep the monitoring up while doing it. This way I know if the healthcheck alerts us it probably means I have genuinely broken allocation

Having said that , I'll need to rewrite the healthcheck to handle the number of variants anyway, so maybe I'll do all that in advance and remove this hack

**Pull request merged**